### PR TITLE
feat: set cursor for view and brushmark

### LINKED ISSUE #5749: Add "cursor" property in config.view schema
I'd like to change the cursor on a chart that allows interval selection to indicate that there's interactivity to it (e.g. `text` or `crosshair`).

When I set it via CSS the cursor already changes outside of the chart area, e.g. on axis labels, where you can make a selection.
I managed to set the cursor to `text` with config.view.cursor:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  ...
  "config": {
    "view": {
      "cursor": "text"
    }
  }
}
```

But with the following warning: `[Warning] Validation: /config/view should NOT have additional properties` . You can try it with this spec: [Example 1](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoAWOgCtY3AHaQANOCgATZAgBOjQnh4qckAOLdu1McgDksMLDzd4AazCF98ZGG5VdYEaoyHRqGpCa6HiYOMCQrLpi2FpR6HzObrB08LAUkAC+6hCQeLroSrAAZty6qMkA2nGVjGJ4yLrJEVGsqHSwAJ6o7NxiALzjtmYA8lNmtgUAukVQGLqunRJKyGHFyErw3JqMStTJoBDFAB5nkM3IYpqdkW1qUHj9hNumbaiENehJQoaYr9G53B6dTyMbyvEofL5QACOrHKRmiRioBQ0QOKsHuyHgRmUZ2BUDKJwR5wuxXen06xzaugoAJ21KgewORxOsAakGuC1J+Wxy0gByUzVOsVJkAojGQAHcSWyoPAEooOt9kJc8JBBdiQEKgA)

I assume the schema does not contain specifying a cursor there, thus this issue :)

~~Bonus question: How would i change the cursor when the mouse is over the interval?~~ (selection._name._mark.cursor (_same warning here_) or config.mark.cursor) : [Example 3](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoAWOgCtY3AHaQANOCgATZAgBOjQnh4qckAOLdu1McgDksMLDzd4AazCF98ZGG5VdYEaoyHRqGpCa6HiYOMCQrLpi2FpR6HzObrB08LAUkAC+6hCQeLroSrAAZty6qMkA2nGVjGJ4yLrJEVGsqHSwAJ6o7NxiALzjtmYA8lNmtgUAukVQGLqunRJKyGHFyErw3JqMStTJoBDFAB5nkM3IYpqdkW1qUHj9hNumbaiENehJQoaYr9G53B6dTyMbyvEofL5QACOrHKRmiRioBQ0QOKsHuyHgRmUZ2BUDKJwR5wuxXen06xzaugoAJ21KgewORxOsAakGuS1JxVW61igoukHgCUUHVMfwZ7UgYvypOVEBxUAOSmap1F1MgFEYyAA7iS2RqpTVOm1LnhFdTVWBlcqgA)

---

I also tried setting the cursor with config.style.cursor.cursor but that didn't work: [Example 2](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoAWOgCtY3AHaQANOCgATZAgBOjQnh4qckAOLdu1McgDksMLDzd4AazCF98ZGG5VdYEaoyHRqGpCa6HiYOMCQrLpi2FpR6HzObrB08LAUkAC+6hCQeLroSrAAZty6qMkA2nGVjGJ4yLrJEVGsqHSwAJ6o7NxiALzjtmYA8lNmtgUAukVQGLqunRJKyGHFyErw3JqMStTJoBDFAB5nkM3IYpqdkW1qUHj9hNumbaiENehJQoaYr9G53B6dTyMbyvEofL5QACOrHKRmiRioBQ0QOKsHuyHgRmUZ2BUDKJwR5wuxXen06xzaugoAJ21KgewORxOsAakGuC1J+Wxy0gByUzVOsVJkCc-RsJLZUHgCUUHSliuKyt0qsh3AZ7UgpIgQupJrAQqFQA). I also wondered why config.style.cursor needs to be an `object`.

### PR DESCRIPTION
**Pull request by keckelt**
Please:

- [x] Make the PR atomic. (Fix one issue at a time.) Multiple relevant issues that must be fixed together? Make atomic commits so we can easily review each issue.

- [x] Provide a concise title so we can easily copy it to the release note.

  - Use imperative mood and present tense.
  - Mention relevant issues. (e.g., `Fix #1` / `Fix part of #1`)

- ~Lint and test (Run `yarn test`)~ (doesnt run in WSL/without chrome)

- [x] Rebase onto the latest `master` branch

- [x] Review your changeset first (to ensure code quality)

- For new features
  - ~Add new unit tests~
  - [x] Update the documentation under `site/docs/` + add examples

---

Fixes #5749 

With this PR, the schema allows to set a cursor on BrushMarks, i.e. for interval selections, and the View.

### CODE REVIEW

**Comment by keckelt** [comment]
TODO:
* [ ] Add cursor property in selection mark documentation: https://vega.github.io/vega-lite/docs/selection-mark.html
* [ ] Add cursor property in view background documentation: https://vega.github.io/vega-lite/docs/spec.html#view-background

File: src/spec/base.ts [hunk 1]
```diff
@@ -145,6 +145,11 @@ export interface BaseViewBackground
    * __Default value:__ `"#ddd"`
    */
   stroke?: Color | null;
+
+  /**
+   * The mouse cursor used over the view. Any valid [CSS cursor type](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor#Values) can be used.
+   */
+  cursor?: Cursor;
```
Comment (domoritz) [inline src/spec/base.ts]: Are we getting a default from Vega here? Are the Vega typings correct? Could you check?

Comment (keckelt) [reply src/spec/base.ts]: The vega-typings are correct, yes:
https://github.com/vega/vega/blob/master/packages/vega-typings/types/spec/config.d.ts#L309-L345

It's an enumeration of the valid strings.

Not sure what you mean with default from Vega. The cursor doesn't change by adding the property at least.

Comment (domoritz) [reply src/spec/base.ts]: We import the Vega typings in many places so often it's not necessary to override property types.

Comment (keckelt) [reply src/spec/base.ts]: There is a cursor definition in `BaseMarkConfig` from which this interface extends, but the documentation would say the property specifies the mouse cursor used over the _**mark**_.

**Comment by domoritz** [comment]
I'll merge with https://github.com/vega/vega-lite/pull/5787 so we can run some more tests. Thanks for your pull request!

**Pull request closed without merging**